### DISCUSSION TITLE
If you could have just one feature/fix added...

### DISCUSSION THREAD

**Discussion by gdbarron** [Snow-Shell/servicenow-powershell#124] [Q&A]
Hi all.  I've just recently taken over as the lead on this project and I'd like to get it moving again.  With that in mind, if you could add just **one** feature or fix, what would it be?  I know 1 is a small number, but let's start with that 😉

**Comment by Konnann78** [comment]
I would like to be able to deploy and reclaim assets to and from users :)

**Comment by wwwb0n3zcom** [comment]
I know that in v3 there are the following tables below. Please add PBI (pbi prefix not found).

inc
chg
req
ritm
sctask
ctask

**Reply by gdbarron** [reply]
Hi @wwwb0n3zcom.  What table/item type is pbi the prefix for?  Please open an [enhancement issue](https://github.com/Snow-Shell/servicenow-powershell/issues/new?assignees=&labels=&template=feature_request.md&title=) with all the details including table/item type and the most important fields so I can add to the format.

**Comment by mgray40** [comment]
The ability to append a short/full description
I have only been able to find documentation on full replacement/creation

**Reply by gdbarron** [reply]
Hey @mgray40.  Could you tell me a bit more about the use case and why you would want to use this over comments/work notes?  Thanks.

**Reply by mgray40** [reply]
We have a number of different job tasks/scripts that in additional to users adding their additional comments or work notes we update the description with a summary of the work completed but also items that need follow up. The ability to have some scripts  append eases the amount of work needed to try formatting everything as a single giant string. 

I only found this module a few days ago and currently the only method I've found to do something similar is to pull the current description, add the new info,  then upload back. Just looking to change the amount of times I need to pull from the existing ticket. 

I am unfamiliar with the Service-Now API, so if that's not something it supports outside of what's above then obviously it doesn't matter

**Reply by gdbarron** [reply]
I've been looking around and don't believe the api supports it.

**Reply by mgray40** [reply]
Thanks for looking!

**Comment by weidi** [comment]
It´d be great really great to get the database model to easier build queries

**Reply by gdbarron** [reply]
Hi @weidi.  Could you expand a bit on what you're looking for?  Does the dictionary not provide it?

**Reply by weidi** [reply]
Hi @gdbarron, i´m quite new to ServiceNow so i´m not aware of a dictionary but from the documentation this sounds good, let me play around a bit and it might be cool to have this in the readme to help others.

**Reply by gdbarron** [reply]
Welcome!  Once you get comfortable with the key tables and fields, I'm sure this module will be very helpful. Have you checked out the community site on servicenow?  They are very helpful and quick to respond, great for when just getting started.

**Comment by walkerpauldavid** [comment]
install, preview and commit update sets from posh would be awesome

**Reply by gdbarron** [reply]
@walkerpauldavid could explain a bit more about this suggestion and how/why the module today doesn't support it and what might be needed to accomplish it?  Thanks.

**Comment by walkerpauldavid** [comment]
Thanks for providing this posh module, struggling to get a complete list of the filter operators, if someone could please advise where to look?

**Reply by gdbarron** [reply]
Hi @walkerpauldavid, here you go
![image](https://user-images.githubusercontent.com/11862024/141799672-d5d6c6dd-5668-4319-94c3-39ca210a5f79.png)

**Comment by RoYoMi** [comment]
Better and more thorough examples to help a newb boot strap into using this module.

**Reply by gdbarron** [reply]
Hi @RoYoMi.  What are some of the scenarios you are looking to automate?

**Reply by RoYoMi** [reply]
I'm sorry for not getting back to you sooner; I was looking for a way to create standard change tickets to communicate environment changes created by automation. This was, however, before the organization I was working for locked out SNow to prevent any automated ticket creation, so the issue largely became moot. 

When I discovered your project, I would like some additional examples of interfacing with SNow to do basic things like ticket creation, updating tickets, and closing tickets by "Tickets," I mean requests, incidents, and change tickets.

**Comment by sbravener** [comment]
I am still a neophyte with both PS and SNOW.  is it possible to pull all tasks associated with a ticket and then close those tasks?  for instance I have a single ticket fed into by a monitor so it may have like 180+ tasks under it from a sensor .  I want to be able to feed PS a ticket # and list all tasks under it and then resolve those tasks in one fell swoop

can this be done with the module as is or is it a future modification?

**Reply by gdbarron** [reply]
Hi @sbravener, welcome to both PS and SNOW :). You need to get the tasks and then close them.  Here's a sample:

`Get-ServiceNowRecord -ParentId ritm0000007 | Update-ServiceNowRecord -InputData @{state = 'Closed Complete'}`

Let me know of any other questions.
